ddl/db_test.go

```diff
@@ -6524,3 +6524,12 @@ func (s *testDBSuite4) TestCreateTableWithDecimalWithDoubleZero(c *C) {
 		checkType("test", "tt", "d")
 	*/
 }
+
+func (s *testDBSuite4) TestIssue22207(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test;")
+	tk.MustExec("set @@session.tidb_enable_table_partition = nightly;")
+	tk.MustExec("create table t1(id char(10)) partition by list columns(id) (partition p0 values in ('a'), partition p1 values in ('b'));")
```

Comment:
Please `drop if exists` the table `t1` and `t2` first or create another database to avoid conflict with other tests.

Reply:
BTW, please also insert some rows to the table to verify the correctness.

Reply:
OK, I have added row `a` to t1, PTAL